utils/githooks/pre-commit.d/20-flake.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/sh
+
+# Runs flake8 for the DAOS project.
+# Will first check uncomiited code, then either the entire tree if on master
+# or a diff against master if not.
+#
+# Picks up config files from .flake8
+
+if ! command -v flake8 > /dev/null 2>&1
+then
+    echo "No flake checking, install flake8 command"
+    exit 0
+fi
+
+echo checking uncommitted code.
+git diff -u | flake8 --diff
+
+RC=$?
+if [ $RC -ne 0 ]
+then
+    exit $RC
+fi
+
+BRANCH=$(git rev-parse --abbrev-ref HEAD > /dev/null 2>&1)
```

Comment:
Should this be error checked with an `if/then/fi` wrapper?

Reply:
Still seems like this would be useful to wrap with an `if` to not confuse the user into thinking they have python code errors if/when this call fails:

```suggestion
if ! BRANCH=$(git rev-parse --abbrev-ref HEAD > /dev/null 2>&1); then
    echo "Failed to determine branch with git rev-parse"
    exit 1
fi
```

Reply:
Done.  Githut automatically marked this as resolved when I accepted your suggestion.

Reply:
> Githut automatically marked this as resolved when I accepted your suggestion.

Interesting.  Doesn't seem unreasonable.  Ideally though, I think github should distinguish between it resolving a conversation because a suggestion was accepted and the PR owner resolving conversations.  But frankly, I think github should allow a project (or maybe it's a per-user preference) to disable allowing anyone except the conversation owner resolving conversations.